### PR TITLE
BLHeli: fixed build without BLHeli with sim on hw

### PR DESCRIPTION
**Pull request by tridge**
this build was failing:
Tools/scripts/sitl-on-hardware/sitl-on-hw.py --board Pixhawk6X --vehicle plane --frame QuadPlane --simclass QuadPlane

### CODE REVIEW

File: libraries/AP_IOMCU/AP_IOMCU.cpp [hunk 1]
```diff
@@ -117,7 +118,7 @@ void AP_IOMCU::thread_main(void)
     uart.begin(1500*1000, 128, 128);
     uart.set_unbuffered_writes(true);
 
-#if HAL_WITH_IO_MCU_BIDIR_DSHOT
+#if HAVE_AP_BLHELI_SUPPORT && HAL_WITH_IO_MCU_BIDIR_DSHOT
```
Comment (andyp1per) [inline libraries/AP_IOMCU/AP_IOMCU.cpp:121]: I'd have to look more closely but I don't think this is correct - I don't think we compile blheli into iomcu

File: libraries/AP_RCTelemetry/AP_Spektrum_Telem.cpp [hunk 2]
```diff
@@ -33,9 +33,7 @@
 #include <AP_Baro/AP_Baro.h>
 #include <AP_RTC/AP_RTC.h>
 #include <AP_SerialManager/AP_SerialManager.h>
-#ifdef HAVE_AP_BLHELI_SUPPORT
-#include <AP_BLheli/AP_BLHeli.h>
```
Comment (andyp1per) [inline libraries/AP_RCTelemetry/AP_Spektrum_Telem.cpp:37]: don't you need a define guard in the header if you do this?

**Review by andyp1per — CHANGES_REQUESTED** [review]
Not sure this is correct, would need to look more closely

File: libraries/AP_IOMCU/AP_IOMCU.cpp (hunk 1 above)
Comment (tridge) [reply libraries/AP_IOMCU/AP_IOMCU.cpp:121]: AP_IOMCU does use AP_BLHeli, and that is the issue. It doesn't build with sim-on-hw

**Comment by tridge** [comment]
@andyp1per AP_BLHeli::telem_data doesn't exist in AP_Spektrum_Telem::calc_esc()
I suspect this was an incomplete changeover in this PR: https://github.com/ArduPilot/ardupilot/pull/16725

**Comment by andyp1per** [comment]
Can't push - here is a fixup - https://github.com/ArduPilot/ardupilot/pull/26554

**Review by andyp1per — CHANGES_REQUESTED** [review]
See my fixup branch

**Comment by tridge** [comment]
@andyp1per thanks!

**Review by andyp1per — APPROVED** [review]

**Pull request merged**